modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateProcessor.java

```diff
@@ -125,6 +125,11 @@ public DateProcessor create(Map<String, Processor.Factory> registry, String proc
                 } catch (IllformedLocaleException e) {
                     throw new IllegalArgumentException("Invalid language tag specified: " + localeString);
                 }
+                for (String part : localeString.split("-")) {
+                    if (part.length() < 2 || part.length() > 3) {
+                        throw new IllegalArgumentException("Invalid language tag specified: " + localeString);
+                    }
+                }
```

Comment:
A single value should still work?

Maybe we could use the validation logic from https://stackoverflow.com/questions/3684747/how-to-validate-a-locale-in-java in order to make sure that the provided locale is recognized by the JVM?